Tasks/IISWebAppManagementOnMachineGroup/Utility.ps1

```diff
@@ -50,12 +48,12 @@ function Validate-Inputs
     )
 
     Write-Verbose "Validating website and application pool inputs"
-    if($createWebsite -ieq "true" -and [string]::IsNullOrWhiteSpace($websiteName))
+    if(($createWebsite -ieq "true" -or $actionIISWebsite -ieq "CreateOrUpdateWebsite") -and [string]::IsNullOrWhiteSpace($websiteName))
     { 
         throw "Website Name cannot be empty if you want to create or update the target website."
     }
 
-    if($createAppPool -ieq "true" -and [string]::IsNullOrWhiteSpace($appPoolName))
+    if(($createAppPool -ieq "true" -or $actionIISApplicationPool -ieq "CreateOrUpdateAppPool") -and [string]::IsNullOrWhiteSpace($appPoolName))
```

Comment:
Isn't validation of inputs not added for other two new commands (example: create application?)